src/main/java/com/turn/ttorrent/common/Torrent.java

```diff
@@ -418,6 +418,12 @@ public void save(OutputStream output) throws IOException {
 		return md.digest();
 	}
 
+	public static byte[] hash(ByteBuffer data) throws NoSuchAlgorithmException {
+		MessageDigest md = MessageDigest.getInstance("SHA-1");
+		md.update(data);
```

Comment:
Only if it's a ByteBuffer which doesn't backed by an accessible byte array, and in that case, only create a 4K buffer, which is more memory friendly solution, than to create a byte buffer at least 64K, to accommodate a torrent block just for hashing. I could run the torrent client more efficiently this way, YMMV.